File: sem4 py/BuyTicket/Ticket.py
from datetime import datetime


class Ticket():
    def __init__(self, rootNumber: int, price: float, place: int = 0):
        self._rootNumber = rootNumber
        self.price = price
        self.place = place
        self.date = date = datetime.today().strftime("%Y.%m.%d")
        self._isValid = False

    def GetRoot(self):
        return self._rootNumber

    def GetTicket(self):
        return [
            f'Ticket №"{self._rootNumber} |Price = {self.price} |Date {self.date} {"|Not Place" if self.place == 0 else "|Place " + str(self.place)} |Status {self._isValid}']

    def SetValidTrue(self):
        self._isValid = True
        return self._isValid

    def GetValid(self):
        return self._isValid

    def GetPrice(self):
        return self.price

###########

File: sem4/src/Ticket.java
import java.util.Date;

public class Ticket {
    public Date date = new Date();
    public long rootNumber;
    public double price;
    public int place = -1;
    public boolean isValid = false;

    public Ticket(long rootNumber, double price, int place) {
        this.rootNumber = rootNumber;
        this.price = price;
        this.place = place;
    }

    public Ticket(int rootNumber, int price) {
        this.rootNumber = rootNumber;
        this.price = price;
    }


    public String getRootNumber() {
        String p = "Not place";
        if (place > 0) {
            p = "Place " +  Integer.toString(place);
        }
        return "Ticket № " + rootNumber + ", Date " + date + ", Price "+ price + " Rub, " + p + "\n";
    }

    public boolean setValid(boolean isValid) {
        this.isValid = isValid;
        return isValid;
    }



}
